packages/yarnpkg-core/sources/scriptUtils.ts

```diff
@@ -230,10 +234,14 @@ export async function prepareExternalProject(cwd: PortablePath, outputPath: Port
               ? [`workspace`, workspace]
               : [];
 
-            // Makes sure that we'll be using Yarn 1.x
-            const version = await execUtils.pipevp(`yarn`, [`set`, `version`, `classic`, `--only-if-needed`], {cwd, env, stdin, stdout, stderr, end: execUtils.EndStrategy.ErrorCode});
```

Comment:
The fix would be to make sure this command is executed using the current version of Yarn, we can't assume which version Corepack is set to use.